cid.go

```diff
@@ -170,42 +175,72 @@ var (
 // Cid represents a self-describing content addressed
 // identifier. It is formed by a Version, a Codec (which indicates
 // a multicodec-packed content type) and a Multihash.
-type Cid struct{ str string }
+type Cid = GenericCid[string]
```

Comment:
We're adding `GenericCid` to the interface - is that the right name? (it feels like it's an implementation detail leaking into the interface name - that it has a generic backing storage should be obvious from it's instantiation)

`CidFrom[string]`?
`CidOf[string]`?
`TypedCid[string]`?

Reply:
I didn't changed `Cid` because of backward compat (this would be gonna bloody).

> We're adding `GenericCid` to the interface - is that the right name? 

Clearly not, looking for something better.
`CidOf` is ok for me.

Reply:
right. i agree with the construction here for backwards compatibility 👍